### ISSUE TITLE
Implement analytics on git (git commit)

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#6896]
**Context:** Tracking a `git commit` is a strong signal that a user is doing something "meaningful" in a workspace, and is useful as the "end" point for our onboarding funnel. A git commit signals that the user has understood things like: Gitpod is a container image, gitpod is mutable and not read-only (e.g. it's not just a https://vscode.dev/ type application), etc. 

---

We currently don't know how/when users are discovering terminal access within the browser-based VS Code. This issue is to implement the potential tracking of terminal events within hosted VS Code. 

Terminal events that are of particular interest are: 

* `git` commands, specifically `git commit` as this tells us when a user has acknowledged that Gitpod is not just a read-only application, and that full git workflows are possible. 
* install commands such as `npm install` or `apt-get` as we want to see if users are understanding that they can codify their installations into the `.gitpod.yml` or not, and whether they're utilising configurations such as prebuilds. 

A technical spike might be required to understand how this could feasibly be implemented. Will also be worth getting the @gitpod-io/engineering-workspace team involved and @csweichel to hep us with this. 

**DISCLAIMER**: We need to be careful with this to respect users privacy. Let's investigate what changes we'll need to make to our privacy policy, or possibly implement an explicit opt-in for users to send us developer data.

**Comment by jeanp413** [comment]
I think we can use this proposed API https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.terminalDataWriteEvent.d.ts

**Comment by loujaybee** [comment]
Spoke with @akosyakov and have reduced the scope on this to be tracking only "git" commands, rather than all terminals.

**Comment by loujaybee** [comment]
Is dependent upon: #7137

**Comment by roboquat** [comment]
@loujaybee: Issue scheduled in the workspace team (WIP: 0)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/issues/6896#issuecomment-993364424):

>/schedule


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by jeanp413** [comment]
From the vscode telemetry:
- there's telemetry for all commands in the git extension, it only logs the git command id, git commands executed from the terminal are not detected.

**Comment by akosyakov** [comment]
We decided to go with tracking git push on credential helper level. Instrumenting git extensions will require maintaining another web socket connection to the server so we rather postpone it.

**Comment by bigint** [comment]
So is this gonna be an anonymous data collection or do you collect user's data (example: ws id)?
